### PR TITLE
Improve autocomplete serialize, deserialize

### PR DESCRIPTION
**Pull request by programmer4web**
_serializeFilter for example if valueProperty is id then in url will be only ids separated by ~
deserializeFilter splits ids and searches in column values for them and returns the items array.

Iulian: The problem is that when we set filter (from deserializeFilter) values might not be set if data property is not set so it can't set selection in paper-autocomplete-chips.

version 1:  Because of that I set the param from url as _pendingFilterString and next time values are set it will update filter.

version 2: ( without flag ) set an array of values in paper-autocomplete-chips. [WIP]

### CODE REVIEW

File: cosmoz-omnitable-column-autocomplete.html [hunk 1]
```diff
@@ -158,6 +158,34 @@
 					}
 					return true;
 				});
+			},
+
+			_serializeFilter: function (obj = this.filter) {
```
Comment (nomego) [inline cosmoz-omnitable-column-autocomplete.html]: - Use es6 syntax and skip `: function `
- Why `obj` (more generic, harder to follow) and not something like just `filter` ?

(same below)

File: cosmoz-omnitable-column-autocomplete.html [hunk 2]
```diff
@@ -158,6 +158,34 @@
...
+			},
+
+			_deserializeFilter: function (obj) {
+				if (obj == null || obj  === '') {
+					return null;
+				}
+				return obj.split('~').map(item => this.values.find(v => {
+					const property = this.valueProperty;
+					if (!property) {
+						const vv = typeof v === 'number' ? v.toString() : v;
```
Comment (nomego) [inline cosmoz-omnitable-column-autocomplete.html]: Why is this needed? Numbers should be able to compare directly, `3 === 3`

File: cosmoz-omnitable-column-autocomplete.html (hunk 1 above)
Comment (megheaiulian) [reply cosmoz-omnitable-column-autocomplete.html]: @nomego Fixed that.

File: cosmoz-omnitable-column-autocomplete.html (hunk 2 above)
Comment (megheaiulian) [reply cosmoz-omnitable-column-autocomplete.html]: @nomego I changed to a `_normalizeValue` function because numbers from hash params are strings sometimes. If possible we convert to numbers.

File: cosmoz-omnitable-column-autocomplete.html [hunk 3]
```diff
@@ -168,7 +171,54 @@
...
+					.map(item => this.values
+						.find(v => this._normalizeValue(property ? this.get(property, v) : v) === this._normalizeValue(item))
+					).filter(i => i != null);
+			},
+
+			_valuesForHashChanged() {
+				if (!Array.isArray(this.values) || this._pendingFilterString) {
+					return;
+				}
+				this.setFilterFromHash(this._pendingFilterString);
```
Comment (nomego) [inline cosmoz-omnitable-column-autocomplete.html]: `_pendingFilterString` can only be falsy? Intended? Where does that property even come from? New in this PR?

Comment (megheaiulian) [reply cosmoz-omnitable-column-autocomplete.html]: @nomego The problem  is that when we set filter (from `deserializeFilter`) `values` might not be set if `data` property is not set so it can't set selection in paper-autocomplete-chips.
Because of that I set the param from url as `_pendingFilterString` and next time values are set it will update filter.

Comment (megheaiulian) [reply cosmoz-omnitable-column-autocomplete.html]: Another solution would be to able to set an array of values in `cosmoz-autocomplete-chips`.
